File: Java/Sorting/TwoSum.java
package sortingPrep;

class Pair<T> {
    public T x;
    public T y;

}

public class TwoSum {
    private static Pair twoSum(int[] arr, int target, Pair pair) {
        for (int i = 0; i < arr.length; i++) {
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[i] + arr[j] == target) {
                    pair.x = i;
                    pair.y = j;
                }
            }
        }
        return pair;
    }

    public static void main(String[] args) {
        int[] arr = { 2, 3, 1, 4, 0, 7, 10, 16, 8, 9, 6 };
        int target = 20;
        Pair pair = new Pair();
        twoSum(arr, target, pair);
        System.out.println("(" + pair.x + "," + pair.y + ")");
    }

}

File: Python/TwoSum.py
def two_sum(arr, target):
    dict = {}
    n = 2
    result = [None] * n
    for i in range(len(arr)):
        diff = target - arr[i]
        if diff in dict:
            result[0] = diff
            result[1] = arr[i]
        dict[arr[i]] = arr[i]
    return result


if __name__ == '__main__':
    arr = [2, 10, 9, 12, 8, 24]
    target = 12
    print(two_sum(arr, target))
